update/process/afterHardFork.go

```diff
@@ -74,22 +74,55 @@ func (a *afterHardFork) CreateAllBlocksAfterHardfork(
 	}
 	shardIDs[a.shardCoordinator.NumberOfShards()] = core.MetachainShardId
 
+	lastPostMbs, err := update.CreateBody(shardIDs, mapBodies, a.mapBlockProcessors)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	err = update.CreatePostBodies(shardIDs, lastPostMbs, mapBodies, a.mapBlockProcessors)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	err = update.CheckDuplicates(shardIDs, mapBodies, a.marshalizer, a.hasher)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	err = a.createHeader(shardIDs, mapBodies, mapHeaders, chainID, round, nonce, epoch)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return mapHeaders, mapBodies, nil
+}
+
+func (a *afterHardFork) createHeader(
+	shardIDs []uint32,
+	mapBodies map[uint32]data.BodyHandler,
+	mapHeaders map[uint32]data.HeaderHandler,
+	chainID string,
+	round uint64,
+	nonce uint64,
+	epoch uint32,
+) error {
 	for _, shardID := range shardIDs {
+		log.Debug("afterHardFork.createHeader", "shard", shardID)
+
 		blockProcessor, ok := a.mapBlockProcessors[shardID]
 		if !ok {
-			return nil, nil, update.ErrNilHardForkBlockProcessor
+			return update.ErrNilHardForkBlockProcessor
 		}
 
-		hdr, body, err := blockProcessor.CreateNewBlock(chainID, round, nonce, epoch)
+		hdr, err := blockProcessor.CreateBlock(mapBodies[shardID], chainID, round, nonce, epoch)
```

Comment:
the remaining miniblocks created by postprocessors has to be added to the genesis metablock - as those are notarized and are waiting to continue processing after hardfork.

Reply:
They are already added in mapBodies. In addition to the created body/miniblocks done in method CreateBody called above also in method CreatePostMiniBlocks line 229, they are appended to the same body.
currentBody.MiniBlocks = append(currentBody.MiniBlocks, postBody.MiniBlocks...)